Browserify customizability enhancements

brettz9 opened the pull request:
- Enhancement: Support `plugins` array (strings, functions, or arrays) in CLI/Node
- Enhancement: `transforms` alias for `transform` to parallel
     `plugins` plural
- Enhancement: Add `entries` alias to parallel `browserify`
- Enhancement: Support custom options to `browserify`
- Enhancement: Return `browserify` instance to support further
    pipelining, etc.
- Enhancement: Support testem options
- Enhancement: Support finalizer function
- Enhancement: Support transform functions
- Fix: Ensure proper testem method is called on watch updates
- Testing: Ensure phantomjs-prebuilt is installed
- Testing: Add tests for missing `files`, not throwing without transforms

Addresses issue #4

brettz9 committed:
- Enhancement: Support `plugins` array (strings, functions, or arrays) in CLI/Node
- Enhancement: `transforms` alias for `transform` to parallel
     `plugins` plural
- Enhancement: Add `entries` alias to parallel `browserify`
- Enhancement: Support custom options to `browserify`
- Enhancement: Return `browserify` instance to support further
    pipelining, etc.
- Enhancement: Support testem options
- Enhancement: Support finalizer function
- Enhancement: Support transform functions
- Fix: Ensure proper testem method is called on watch updates
- Testing: Ensure phantomjs-prebuilt is installed
- Testing: Add tests for missing `files`, not throwing without transforms

alekseykulikov commented:
@brettz9 thank you for PR! Could you add tests for this functionality? and I will merge.

brettz9 commented:
I've added tests. I know you're a fan of keeping down the features per PR, but I added a couple others related to customizability, some of which helped ensure the testing worked properly.

brettz9 commented:
Also, FWIW, I implemented #3 and the tests from #2 (which had already been implemented).

alekseykulikov commented:
hey @brettz9 
thank you for good work, a few notes for future enhancement:
- I need to release v3 (need to update deps)
- remove testem output from `npm test` output
- maybe it makes sense to separate cli.js and index.js tests, or just make all tests to run with `exec` (I primary use this library as cli)

alekseykulikov commented:
I've also added you as a collaborator in case you have other ideas for improvement ;)

brettz9 commented:
Thanks for the access.

I've updated the dependencies (using [`ncu`](https://www.npmjs.com/package/npm-check-updates)).

As far as suppressing testem output, yes, I think we ought to be able to do that by supplying testem `reporter` config.

My current reliance on `JSON.parse` (as opposed to say `eval`) doesn't allow functions to be supplied as config, so in that regard the cli is more limited and couldn't support the tests using functions.

alekseykulikov commented:
I also did a few improvements: separate cli/index tests, removed babel-eslint (because of new eslint), made issue for testem output #7
